Tools/CISupport/ews-build/steps.py

```diff
@@ -7084,43 +7092,169 @@ def run(self):
         self.log_observer = logobserver.BufferLogObserver()
         self.addLogObserver('stdio', self.log_observer)
 
+        self.log_observer_json = logobserver.BufferLogObserver()
+        self.addLogObserver('json', self.log_observer_json)
+
         rc = yield super().run()
         if rc != SUCCESS:
             return defer.returnValue(rc)
 
-        self.createResultMessage()
+        self.find_unexpected_results()
+        unexpected_results = self.getProperty('num_failing_files', 0) or self.getProperty('num_unexpected_issues', 0) or self.getProperty('num_passing_files', 0)
+        unexpected_results_after_filter = None
+        if self.expectations and unexpected_results:  # Only consult results database on the first run
+            logTextJson = self.log_observer_json.getStdout()
+            yield self._addToLog('stdio', f'Checking results database for unexpected results...\n')
+            successful_filter = yield self.filter_results_using_results_db(logTextJson)
+            unexpected_results_after_filter = self.getProperty('num_failing_files', 0) or self.getProperty('num_unexpected_issues', 0) or self.getProperty('num_passing_files', 0)
```

Comment:
It does set properties. The reason I did this was because `DisplaySaferCPPResults` uses these properties, so I thought they could be used intra-step since we're setting them anyways